src/crimson/osd/replicated_recovery_backend.cc

```diff
@@ -430,7 +430,11 @@ void ReplicatedRecoveryBackend::prepare_pull(
   pg_missing_tracker_t local_missing = pg.get_local_missing();
   const auto missing_iter = local_missing.get_items().find(soid);
   auto m = pg.get_missing_loc_shards();
-  pg_shard_t fromshard = *(m[soid].begin());
+  assert(m.contains(soid));
+  auto locs = m.at(soid);
+  auto iter = locs.begin();
+  assert(iter != locs.end());
+  pg_shard_t fromshard = *(iter);
```

Comment:
Please see: https://github.com/ceph/ceph/pull/56611/.

Reply:
Um, it seems that #56611 is irrelevant to this commit.